iter/iter.go

```diff
@@ -9,28 +9,64 @@ import (
 	"github.com/sourcegraph/sourcegraph/lib/errors"
 )
 
+// defaultMaxGoroutines returns the default maximum number of
+// goroutines to use within this package.
+func defaultMaxGoroutines() int { return runtime.GOMAXPROCS(0) }
```

Comment:
I had originally thought to additionally provide the option to set a package-level default when I moved this default here, but figured it might be out of scope here

That said, I think that it is surprising to have a default limit at all - I would assume `iter` to be unbounded by default. The default is worth a discussion IMO, but I think would be a separate discussion

Reply:
Yeah, I'm definitely torn on this one. I'll open up an issue and move discussion there.